atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -276,28 +278,22 @@ private CassandraServer getAddressForHostThrowUnchecked(String host) {
 
     @VisibleForTesting
     CassandraServer getAddressForHost(String inputHost) throws UnknownHostException {
-        InetSocketAddress cassHostAddress = config.addressTranslation().containsKey(inputHost)
-                ? config.addressTranslation().get(inputHost)
-                : new InetSocketAddress(InetAddress.getByName(inputHost), getKnownPort());
-
-        Set<CassandraServer> knownNodes = getAllKnownServers();
-
-        for (CassandraServer server : knownNodes) {
-            if (Objects.equals(server.cassandraHostAddress().getAddress(), cassHostAddress.getAddress())) {
-                // todo(snanda): Remove the log if algorithm changes
-                Set<InetSocketAddress> reachableProxies = ImmutableSet.copyOf(getReachableProxies(inputHost));
-                if (!reachableProxies.equals(ImmutableSet.copyOf(server.reachableProxyIps()))) {
-                    log.warn(
-                            "List of proxies has changed",
-                            SafeArg.of("oldProxies", server.reachableProxyIps().toString()),
-                            SafeArg.of("newProxies", reachableProxies.toString()),
-                            SafeArg.of("cassandraHost", inputHost));
-                }
-                return server;
-            }
-        }
+        Map<String, String> hostnamesByIp = hostnameByIpSupplier.get();
+        String cassandraHostName = hostnamesByIp.getOrDefault(inputHost, inputHost);
+        return CassandraServer.builder()
+                .cassandraHostName(cassandraHostName)
+                .addAllReachableProxyIps(getReachableProxiesThrowUnchecked(cassandraHostName))
+                .build();
+    }
 
-        return getCassandraServer(cassHostAddress);
+    private List<InetSocketAddress> getReachableProxiesThrowUnchecked(String inputHost) throws UnknownHostException {
```

Comment:
what does the "throw unchecked" mean here?

Reply:
err i killed the other method so this makes no sense now, will rename